### PR TITLE
➕ Add Code Inspector rules section

### PR DESCRIPTION
**Pull request by cairoeth**

### CODE REVIEW

File: docs/modules/ROOT/pages/module/code.adoc [hunk 1]
```diff
@@ -120,4 +119,436 @@ image::code-settings-advanced.png[Code Advanced Settings]
...
+|Identifies usage of chainlink's deprecated functions.
+|medium
+
+|`check-consistent-usage-of-msgsender-msgdata`
+|Identifies usage of `msg.sender` or `msg.data` when `_msgSender()` or/and `_msgData()` are present
+|note
+
+|`check-effect-interact`
+|Identifies a possible violation of the check, effect and interact pattern.
+|ethtrust
```
Comment (ggonzalez94) [inline docs/modules/ROOT/pages/module/code.adoc:153]: Do we actually display ethtrust as a severity in the UI? I don't remember having seen this in any of my reports

File: docs/modules/ROOT/pages/module/code.adoc [hunk 2]
```diff
@@ -120,4 +119,436 @@ image::code-settings-advanced.png[Code Advanced Settings]
...
+
+|`check-erc4337-compatibility`
+|Identifies if the contract may not be compatible with ERC-4337.
+|note
+
+|`check-return-data-from-external-call`
+|Identifies when the external call return data check is missing.
+|note
+
+|`conflicting-inheritance`
```
Comment (ggonzalez94) [inline docs/modules/ROOT/pages/module/code.adoc:163]: I see this rule and others are marked as not-defender in the [contract inspector docs](https://github.com/OpenZeppelin/contract-inspector/blob/main/docs/rule_ids.md). Does that mean that they are not included for Defender?

**Review by ggonzalez94 — APPROVED** [review]
LGTM. Just a couple questions

File: docs/modules/ROOT/pages/module/code.adoc (hunk 1 above)
Comment (cairoeth) [reply docs/modules/ROOT/pages/module/code.adoc:153]: yep there's a label

File: docs/modules/ROOT/pages/module/code.adoc (hunk 2 above)
Comment (cairoeth) [reply docs/modules/ROOT/pages/module/code.adoc]: thanks! fixed

**Pull request merged**